### ISSUE TITLE
Processors (All or some of them) will consume atleast a bit of energy power

### ISSUE THREAD

**Issue by yozozchomutova** [Anuken/Mindustry-Suggestions#3510]
### Describe what you would like changed, and why.

Processors can help a lot in some situations and I think it would be great (and more realistic) to atleast consume some energy and not just operate for free. 
How much?
-I would let it decide on others/developers. 

### Describe the changes you want to propose. Include possible alternatives.

All or some of 3 processors will be consuming atleast some power, when operating.

**Comment by RealYeet** [comment]
I see many, many, many broken schematics if this gets added.

**Comment by RealYeet** [comment]
In many schematics every block counts, if you have extra space for one more node, you should retire as a schematic designer. 

> > I see many, many, many broken schematics if this gets added.
> 
> Yea, but fix should be easy... Just add one power mode (Sometimes maybe more than one)

**Comment by guncatt** [comment]
I like it logic is a very strong tool.

**Comment by end-me-please** [comment]
Doing calculations isn’t actually doing any significant work.
conveyors also don’t need power.

Turrets that generate huge laser beams do need power.

**Comment by itcannotbe** [comment]
> Doing calculations isn’t actually doing any significant work. conveyors also don’t need power.
> 
> Turrets that generate huge laser beams do need power.

TL;DR:

Mindustry isn't aiming to be a realistic game.
Also Realism ≠ Fun/Entertaining

**Comment by itcannotbe** [comment]
Also adding a power requirement to "nerf" something have never worked for mindustry.

**Comment by yozozchomutova** [comment]
1.
> In many schematics every block counts, if you have extra space for one more node, you should retire as a schematic designer.
> 
> > > I see many, many, many broken schematics if this gets added.
> > 
> > 
> > Yea, but fix should be easy... Just add one power mode (Sometimes maybe more than one)

Yes, that's true... Welp, if it passes, people may have to get again creative to redesign some of their schematics... (or create completely new)

2.
> Doing calculations isn’t actually doing any significant work. conveyors also don’t need power.
> 
> Turrets that generate huge laser beams do need power.

We're not speaking just only about calculations. We're speaking about whole processors, which helps you alot, if you know how to use them. (Especially controlling units)


3.
> Also adding a power requirement to "nerf" something have never worked for mindustry.

It depends on amount of power requirement...

**Comment by itcannotbe** [comment]
> > Also adding a power requirement to "nerf" something have never worked for mindustry.
> 
> It depends on amount of power requirement...

No it's not, power is easy to make.

**Comment by numer99** [comment]
it most likely is that they can generate their own power, like conveyors and lower tier drills

**Comment by yozozchomutova** [comment]
> > > Also adding a power requirement to "nerf" something have never worked for mindustry.
> > 
> > 
> > It depends on amount of power requirement...
> 
> No it's not, power is easy to make.

If you have lot of coal, magma, water, scrap (impact reactor), then yes. But If you don't have these and you're limited with energy power, then you have to decide which energy-consuming buildings are important to you...

> it most likely is that they can generate their own power, like conveyors and lower tier drills

That makes sense, but issue is, that it can be too Op and it should receive some nerf.

**Comment by guncatt** [comment]
I do think small logic processer does need a nerf almost nobody uses the higher tier ones.

**Comment by IxGamerXL** [comment]
It would be annoying to connect everything to power depending on the schematics, with Logic Art being the most power intensive.

**Comment by numer99** [comment]
most people do not use logic

**Comment by itcannotbe** [comment]
> a lot of people don't even use logic

actually a lot of people just steal other people made logic

**Comment by numer99** [comment]
also, what would happen if you ran out of power?
would the logic just stop working?
there would be many problems in the coding of the game, and the coding of the logic blocks
and people would just make votes against it anyway

**Comment by yozozchomutova** [comment]
> also, what would happen if you ran out of power? would the logic just stop working? there would be many problems in the coding of the game, and the coding of the logic blocks and people would just make votes against it anyway

Good question. As others buildings, it slows command-execution speed. Or it can just completely stop and resumes, when processor got full power again...

**Comment by numer99** [comment]
that would cause way too many problems, and make a lot a coding useless
example: emergency power reserves
example: energy-free schematics
